src/org/jitsi/impl/neomedia/rtp/remotebitrateestimator/RemoteBitrateEstimatorSingleStream.java

```diff
@@ -173,81 +173,14 @@ public long getTimeUntilNextRun()
-        if (it == null)
-        {
-            // This is a new SSRC. Adding to map.
-            // TODO(holmer): If the channel changes SSRC the old SSRC will still
-            // be around in this map until the channel is deleted. This is OK
-            // since the callback will no longer be called for the old SSRC.
-            // This will be automatically cleaned up when we have one
-            // RemoteBitrateEstimator per REMB group.
-            it = new Detector(nowMs, new OverUseDetectorOptions(), true);
-            overuseDetectors.put(ssrc_, it);
-            ssrcs = null;
-        }
-
-        // XXX The variable naming is chosen to keep the source code close to
-        // the original.
-        Detector estimator = it;
-
-        estimator.lastPacketTimeMs = nowMs;
-        this.incomingBitrate.update(pkt.getPayloadLength(), nowMs);
-
-        BandwidthUsage priorState = estimator.detector.getState();
-        long[] deltas = this.deltas;
-
-        /* long timestampDelta */ deltas[0] = 0;
-        /* long timeDelta */ deltas[1] = 0;
-        /* int sizeDelta */ deltas[2] = 0;
-
-        if (estimator.interArrival.computeDeltas(
-                pkt.getTimestamp(),
-                System.currentTimeMillis(),
-                pkt.getPayloadLength(),
-                deltas))
-        {
-            double timestampDeltaMs
-                = /* timestampDelta */ deltas[0] * kTimestampToMs;
-
-            estimator.estimator.update(
-                    /* timeDelta */ deltas[1],
-                    timestampDeltaMs,
-                    /* sizeDelta */ (int) deltas[2],
-                    estimator.detector.getState());
-            estimator.detector.detect(
-                    estimator.estimator.getOffset(),
-                    timestampDeltaMs,
-                    estimator.estimator.getNumOfDeltas(),
-                    nowMs);
-        }
-        if (estimator.detector.getState() == BandwidthUsage.kBwOverusing)
-        {
-            long incomingBitrateBps = this.incomingBitrate.getRate(nowMs);
-
-            if (priorState != BandwidthUsage.kBwOverusing
-                    || remoteRate.isTimeToReduceFurther(
-                            nowMs,
-                            incomingBitrateBps))
-            {
-                // The first overuse should immediately trigger a new estimate.
-                // We also have to update the estimate immediately if we are
-                // overusing and the target bitrate is too high compared to what
-                // we are receiving.
-                updateEstimate(nowMs);
-            }
-        }
-        } // synchronized (critSect)
+    public RawPacket reverseTransform(RawPacket pkt){
+        /**
+         * @TODO Findout what to do with the interArrivalTime in this method
+         */
+        long interArrivalTime = getInterArrivalDelayVariation (pkt.getSSRC(),
+        pkt.getTimestamp(),
+        System.currentTimeMillis(),
+        pkt.getLength());
```

Comment:
This used to be getPayloadLength(), but I'm not sure what we want. I think if you're going to use the value as an encoder's target bitrate (as in webrtc.org) then you want the payload length only. But our bitrate calculations include RTP headers.
cc @gpolitis